addons/repair/models/repair.py

```diff
@@ -145,57 +144,48 @@ def _compute_allowed_picking_type_ids(self):
         self.allowed_picking_type_ids = [
             pt['return_picking_type_id'] for pt in out_picking_types if pt['return_picking_type_id']]
 
-    @api.depends('partner_id')
-    def _compute_default_address_id(self):
-        for order in self:
-            if order.partner_id:
-                order.default_address_id = order.partner_id.address_get(['contact'])['contact']
-
-    @api.depends('pricelist_id', 'company_id')
-    def _compute_currency_id(self):
-        for order in self:
-            order.currency_id = order.pricelist_id.currency_id.id or order.company_id.currency_id.id
+    @api.depends('parts_availability_state')
+    def _compute_availability_boolean(self):
+        self.is_parts_available, self.is_parts_late = False, False
+        repairs = self.filtered(lambda repair: repair.parts_availability_state)
+        repairs.filtered(lambda repair: repair.parts_availability_state == 'available').write({'is_parts_available' : True})
```

Comment:
Don't explicit write in compute. It could act like an onchange and if the user discard it we won't rollback the modification. Instead use a direct assignation